backend/server/core/services/calculateAchievementsStatus.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Achievement } from '../interfaces/achievement.interface';
+import { Status } from '../interfaces/status.interface';
+
+/**
+ * Returns achievements statusfor the challenge by its achievements listand tasks status
+ * @param achievements - a list of achievements
+ * @param taskStatus - tasks status
+ */
+// type CalculateAchievementStatus = (
+//     achievements: Achievement[],
+//     tasksStatus: Status
+//   ) => Status;
+
+export function calculateAchievementsStatus(
+  achievements: Achievement[],
+  tasksStatus: Status
+): Map<number, Status> {
+  const statuses: Map<number, Status> = new Map();
+
+  achievements.forEach((achievement) => {
+    statuses.set(achievement.id, tasksStatus);
+  });
+
```

Comment:
We have a challenge with default duration 30 days. We have some achievements for this period (30 / 6 = 5 achievements per month by default). Every achievement has some tasks (6 tasks by default).

There are 2 statuses for an achievement ("In Progress" and "Done"). "In Progress" means user hasn't finished yet all the tasks in the achievement. "Done" means user has finished all the tasks already.
So the function should calculate status for each achievement by checking its tasks statuses.